### ISSUE TITLE
showHelp() shows incorrect option list

### ISSUE THREAD

**Issue by zaro** [bcoe/yargs#128]
I just migrated several scripts from optimist to yargs and I noticed that yargs omits some options when generating help. Here is an example :

```javascript
var opts= require('yargs')
        .usage(
                "Usage: $0 [options]"
        )
        .describe('createdAfter','CreatedAfter ListOrders parameter')
        .alias('a','createdAfter')
        .describe('createdBefore','CreatedBefore ListOrders parameter')
        .alias('b','createdBefore')
        .describe('lastUpdatedAfter','LastUpdatedAfter ListOrders parameter')
        .alias('u','lastUpdatedAfter')
        .describe('lastUpdatedBefore','LastUpdatedBefore ListOrders parameter')
        .alias('v','lastUpdatedBefore')
        .describe('incremental','Perform incremental update based on last synccessfull sync')
        .alias('i','incremental')
        .default('i',false)
        .boolean('i')
        .describe('force','Force recheck')
        .alias('f','force')
        .default('f',false)
        .boolean('f')
        .describe('description','Sync transaction description')
        .alias('d','description')
        .default('d',"#{ require('os').hostname() }:#{ process.title } - #{ process.pid }")
        .string('d');
var argv = opts.argv;

opts.showHelp();
```

This outputs :
```
Usage: example.js [options]

Options:
  -i, --incremental                                             [default: false]
  -f, --force                                                   [default: false]
  -d, --description  
[default: "#{ require('os').hostname() }:#{ process.title } - #{ process.pid }"]
```
As you can see the first 4 options are missing.

**Comment by bcoe** [comment]
@zaro good find. I tend to use the `option` method which allows you to create an alias and description at the same time -- I will put a test case around this use case and get you a patch tomorrow hopefully.

**Comment by bcoe** [comment]
@zaro FYI, it appears to be something silly with the usage logic; when building up a list of parameters to print, it runs into trouble if you define aliases as `alias -> described-key` rather than `described-key -> alias`, a quick fix is this:

```javascript
var opts= require('yargs')
        .usage(
                "Usage: $0 [options]"
        )
        .describe('createdAfter','CreatedAfter ListOrders parameter')
        .alias('createdAfter', 'a')
        .describe('createdBefore','CreatedBefore ListOrders parameter')
        .alias('createdBefore', 'b')
        .describe('lastUpdatedAfter','LastUpdatedAfter ListOrders parameter')
        .alias('lastUpdatedAfter', 'u')
        .describe('lastUpdatedBefore','LastUpdatedBefore ListOrders parameter')
        .alias('lastUpdatedBefore', 'v')
        .describe('incremental','Perform incremental update based on last synccessfull sync')
        .alias('incremental', 'i')
        .default('i',false)
        .boolean('i')
        .describe('force','Force recheck')
        .alias('f','force')
        .default('f',false)
        .boolean('f')
        .describe('description','Sync transaction description')
        .alias('d','description')
        .default('d',"#{ require('os').hostname() }:#{ process.title } - #{ process.pid }")
        .string('d');
var argv = opts.argv;

opts.showHelp();
```

But let's patch this confusion.

**Comment by bcoe** [comment]
@zaro could you give this branch a try:

https://github.com/bcoe/yargs/pull/135

and let me know if it solves your problem?

**Comment by bcoe** [comment]
@zaro  this should now be fixed in `yargs@3.7.0`,

`npm install yargs@3.7.0 --save`.

**Comment by zaro** [comment]
got some time to test this just now.

works great , thanks :)
